clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -539,20 +539,22 @@ public boolean isUpdateDue(long now) {
         public long maybeUpdate(long now) {
             // should we update our metadata?
             long timeToNextMetadataUpdate = metadata.timeToNextUpdate(now);
-            long timeToNextReconnectAttempt = Math.max(this.lastNoNodeAvailableMs + metadata.refreshBackoff() - now, 0);
```

Comment:
Just so I understand the effect of this. Previously we would ensure that we always backoff for at least the metadata refresh backoff if we ever hit the case where no node is currently available. With this patch, the new behavior is that we'll try again no later than the reconnect backoff. We may try again much sooner, but `ClusterConnectionStates` will prevent us from reconnecting to any particular node within the reconnect backoff window. Is that right?

By the way, seems like we can now remove `Metadata.refreshBackoff`?